### PR TITLE
Imp/add componentDidCatch()

### PR DESCRIPTION
**Pull request by billyxs**
- fix isRefreshing waiter updates
- add componentDidCatch() to prevent waiter false reject state

### CODE REVIEW

File: src/connectWaiter.js [hunk 1]
```diff
@@ -68,6 +68,10 @@ export default (params) => {
         }
       }
 
+      componentDidCatch(error) {
+        console.log('redux-waiter: We caught an error in your component. It was caught to prevent the waiter from rejecting your request.')
```
Comment (billyxs) [inline src/connectWaiter.js]: Let me know what you think of this. Been looking forward to adding this false "reject" to the promise when the internal component blows up.

File: src/connectWaiter.js [hunk 2]
```diff
@@ -68,6 +68,10 @@ export default (params) => {
         }
       }
 
+      componentDidCatch(error) {
```
Comment (billyxs) [inline src/connectWaiter.js]: @seesleestak let me know what you think of all this. Been looking forward to using it to stop false "reject" states when the wrapped component throws an error.

Comment (seesleestak) [reply src/connectWaiter.js]: Besides logging the warning, do we have to do anything to handle the `error`?

Also, I'm seeing this from eslint -- which I imagine we can disable if it's not needed:
```
class-methods-use-this: Expected 'this' to be used by class method 'componentDidCatch'.                                                                     
```

File: src/waiter-reducer.js [hunk 3]
```diff
@@ -46,6 +46,8 @@ const reducerMap = {
       name,
     } = payload
 
+    // If we alredy have a response
+    // we are in refresh mode
     const isRefreshing = !!state.response
```
Comment (seesleestak) [inline src/waiter-reducer.js]: Interesting. I'd never really thought about `isRefreshing`. Thanks for the clarification here.

File: __tests__/connectWaiter.test.js [hunk 4]
```diff
@@ -147,5 +147,48 @@ describe('(Redux Waiter) Connect Waiter', () => {
...
+      )
+    })
+
+    it('Should not reject request with connectWaiter inner component error', async () => {
+      expect.assertions(9)
+      const wrapper = mountComponent(Component)
+
+      // component should be stable when an error is thrown inside
+      const error = new Error('test');
+      wrapper.find('Component').simulateError(error);
```
Comment (seesleestak) [inline __tests__/connectWaiter.test.js]: I think I need to spend some time in the enzyme docs :)

File: src/connectWaiter.js (hunk 2 above)
Comment (seesleestak) [reply src/connectWaiter.js]: I'm also seeing it complain about the unused `error`

Comment (billyxs) [reply src/connectWaiter.js]: This is meant to safeguard the waiter request lifecyle, so I don't think connectWaiter should handle the error. Leave that to user implementing the component. I'll see about fixing the lint errors

File: src/connectWaiter.js [hunk 5]
```diff
@@ -1,4 +1,8 @@
+// For componentDidCatch as we just want to warn of an issue
+// for a user's wrapped component when an erro is thrown
```
Comment (seesleestak) [inline src/connectWaiter.js]: "erro" :smiley:

**Pull request merged**